lib/SecretsCrypto/sf-secrets-collection.h

```diff
@@ -0,0 +1,68 @@
+#ifndef SF_SECRETS_COLLECTION_H
+#define SF_SECRETS_COLLECTION_H
+
+#include <glib-object.h>
+#include <gio/gio.h>
+
+#include <sf-secrets-secret.h>
+
+#define SF_TYPE_SECRETS_COLLECTION (sf_secrets_collection_get_type())
+#define SF_SECRETS_COLLECTION(o) (G_TYPE_CHECK_INSTANCE_CAST((o), SF_TYPE_SECRETS_COLLECTION, SfSecretsCollection))
+#define SF_IS_SECRETS_COLLECTION(o) (G_TYPE_CHECK_INSTANCE_TYPE((o), SF_TYPE_SECRETS_COLLECTION))
+
+typedef struct SfSecretsCollection_ SfSecretsCollection;
+typedef struct SfSecretsCollectionClass_ SfSecretsCollectionClass;
+typedef enum SfSecretsCollectionFlags_ SfSecretsCollectionFlags;
+
+enum SfSecretsCollectionFlags_ {
+	SF_SECRETS_COLLECTION_FLAGS_MODE_GET = 0 << 0,
+	SF_SECRETS_COLLECTION_FLAGS_MODE_CREATE = 1 << 0,
+	SF_SECRETS_COLLECTION_FLAGS_MODE_ENSURE = 2 << 0,
+	SF_SECRETS_COLLECTION_FLAGS_MODE_DEFAULT = 3 << 0,
+	SF_SECRETS_COLLECTION_FLAGS_MODE_MASK = 3 << 0,
+};
+
+struct SfSecretsCollection_ {
+	GObject parent;
+};
+
+struct SfSecretsCollectionClass_ {
+	GObjectClass parent_class;
+};
+
+GType sf_secrets_collection_get_type(void);
+
+const gchar *sf_secrets_collection_get_plugin_name(SfSecretsCollection *collection);
+const gchar *sf_secrets_collection_get_encryption_plugin_name(SfSecretsCollection *collection);
+const gchar *sf_secrets_collection_get_name(SfSecretsCollection *collection);
+
+void sf_secrets_collection_get_secret(SfSecretsCollection *collection,
```

Comment:
In the C++ API, collections are exposed merely as data, while operations are performed via the Request APIs.  In this API, it seems like the collection is more "alive" (in that each instance somehow has a connection to the manager, and we have methods like these which get/set/delete secrets from a collection, passing in the collection itself rather than just its identifier).

What are the advantages of doing it this way, as opposed to keeping the API more symmetric / consistent with the C++ API?

Reply:
My though process was along the lines that a user of the API would be more likely to put several secrets in a single collection, it'd be a logical unit to have, and a logical split between source files too.

And also kind-of aiming for "if the collection object exists, the collection exists", although that cannot be guaranteed with the D-Bus API.

Reply:
Certainly I agree with the source code being split into different files.  I do understand the argument for the collection being the interaction endpoint (e.g. collection.save(secret), etc) from a "notional" standpoint, however there are (IMO) significant cognitive costs associated with that decision from an API perspective.  You noted one example of that above.

In general, my preference would be to mirror the C++ API as much as possible, but I am of course open to changing my mind if there are strong reasons to do otherwise in the C API context.

Reply:
My goal was to write an API that I would personally like to use, but don't have strong opinion one way or another.

Changing it should be relatively low effort, much of the code can be used almost as-is.

Reply:
If you write an API that you personally like, if guarantees that somebody is going to like it (i.e. at least yourself). But if you write an API that even you don't like, it doesn't guarantee anything and it may end up being an API that no one likes!

Reply:
+1 for consistency with the C++ API.